nemo/core/neural_modules.py

```diff
@@ -37,47 +38,137 @@ class WeightShareTransform(Enum):
 
 class NeuralModule(ABC):
     """Abstract class that every Neural Module must inherit from.
-
-    Args:
-        pretrained_model_name (str): name of pretrained model to use in order
-            to initialize this neural module
-        factory (NeuralModuleFactory): :class:`NeuralModuleFactory` which
-            created or which should mange this instance. Required for
-            multi-gpu training.
-        placement (DeviceType): (default:None) where this module should
-            be placed. If provided, this parameter takes precedence over
-            whatever is specified in factory.
     """
 
-    def __init__(
-        self, *, pretrained_model_name=None, factory=None, placement=None, **kwargs,
-    ):
-        self._pretrained_model_name = pretrained_model_name
-        self._local_parameters = self.update_local_params()
+    def __init__(self):
+
+        # Get default factory.
+        self._factory = NeuralModuleFactory.get_default_factory()
 
-        default_factory = NeuralModuleFactory.get_default_factory()
-        if (factory is None) and (default_factory is not None):
-            factory = default_factory
+        # TK: So starting from that point - is it even possible that the factory will be None??
 
         # Set module properties from factory else use defaults
-        self._placement = factory.placement if factory is not None else DeviceType.GPU
-        self._opt_level = factory.optim_level if factory is not None else Optimization.mxprO0
+        self._placement = self._factory.placement
+        # If one needs to change that should override it manually.
 
-        # Update module properties using overrides if overrides exist
-        if placement is not None:
-            self._placement = placement
+        # Optimization level.
+        self._opt_level = self._factory.optim_level
 
-        self._factory = factory
+        # Get object UUID.
         self._uuid = str(uuid.uuid4())
 
-        # if kwargs:
-        #    nemo.logging.warning(
-        #        "When constructing {}. The base "
-        #        "NeuralModule class received the following unused "
-        #        "arguments:".format(self.__class__.__name__))
-        #    nemo.logging.warning("{}".format(kwargs.keys()))
+        # Retrieve dictionary of parameters (keys, values) passed to init.
+        self._init_params = self.extract_init_params()
+
+        # Pint the types of the values.
+        # for key, value in self._init_params.items():
+        #    print("{}: {} ({})".format(key, value, type(value)))
+
+        # Validate the parameters.
+        # self.validate_params(self._init_params)
+
+    def extract_init_params(self):
```

Comment:
I think this should be Python "private" method - e.g. "__extract_init_params"

Reply:
ok

Reply:
Do you want this to be the new "rule of thumb" that we will use starting  from now?

Asking as haven't seen any consistency in this matter, **update_local_params()** being the first example...